docs/contracts.rst

```diff
@@ -831,11 +831,90 @@ The general inheritance system is very similar to
 `Python's <https://docs.python.org/3/tutorial/classes.html#inheritance>`_,
 especially concerning multiple inheritance.
 
+Declaring two elements (for example functions) of the same name in
+two different contracts that are part of the same inheritance hierarchy,
+is called _overriding_. Declaring two elements of the same name in the
+same contract is called _overloading_.
+
+If a function is callable on a contract, it must also be callable in the derived contract.
+This means that visibility of functions cannot be restricted, but it
+can be extended. Mutability in turn, can be restricted, but not extended.
+
+The visibility can change from external to public, but not from public
+to external or from external to internal. Functions that are marked ``view`` in the
+super contract, can be marked ``pure`` in the derived contract, but not the other
+way around:
+
++---------------------+------------------------------+
+| A function          | ... can be overridden by a   |
+| with visibility ... | function with visibility ... |
++=====================+==============================+
+| ``external``        | ``external`` or ``public``   |
++---------------------+------------------------------+
+| ``public``          | ``public``                   |
++---------------------+------------------------------+
+| ``internal``        | ``internal`` or ``public``   |
++---------------------+------------------------------+
+| ``private``         | (none)                       |
++---------------------+------------------------------+
+
++---------------------+---------------------------------+
+| A function          | ... can be overridden by a      |
+| with mutability ... | function with mutability ...    |
++=====================+=================================+
+| ``payable``         | ``payable``                     |
```

Comment:
I am not yet clear about this rule.

Reply:
A payable function should be able to be overridden by a function with mutability default, view or pure. You can this already today using boilerplate stubs, see: https://github.com/ethereum/solidity/issues/3412

Also, this rule is assumed by ERC-721.

Reply:
yes, that's probably right. @axic what do you think?